src/llvm/basic_block_collection.cr

```diff
@@ -16,4 +20,32 @@ struct LLVM::BasicBlockCollection
     yield builder
     block
   end
+
+  def each : Nil
+    bb = LibLLVM.get_first_basic_block(@function)
+    while bb
+      yield LLVM::BasicBlock.new bb
+      bb = LibLLVM.get_next_basic_block(bb)
+    end
+  end
+
+  def []?(name : String)
+    find(&.name.==(name))
+  end
+
+  def [](name : String)
+    self[name]? || raise IndexError.new
+  end
+
+  def last?
+    block = nil
+    each do |current_block|
+      block = current_block
+    end
+    block
+  end
+
+  def alloca_block?
+    self["alloca"]?
+  end
```

Comment:
Is this used anywhere?

Reply:
It used to be, not more but can be good for someone who may write their own compiler by the means of Crystal

Reply:
Please remove all methods which aren't used.

Reply:
ok